libraries/AP_HAL_ChibiOS/hwdef/ARK_PI6X/README.md

```diff
@@ -0,0 +1,156 @@
+| 5   | I2C1_SDA_GPS1_EXT   | 3.3V    |
+| 6   | GND                 | GND     |
+
+#### TELEM1 - 6 Pin JST-GH
+| Pin | Signal Name           | Voltage |
+|-----|-----------------------|---------|
+| 1   | 5.0V                 | 5.0V    |
+| 2   | UART7_TX_TELEM1_EXT  | 3.3V    |
+| 3   | UART7_RX_TELEM1_EXT  | 3.3V    |
+| 4   | UART7_CTS_TELEM1_EXT | 3.3V    |
+| 5   | UART7_RTS_TELEM1_EXT | 3.3V    |
+| 6   | GND                  | GND     |
+
+#### Flight Controller Debug - 10 Pin JST-SH
+| Pin | Signal Name     | Voltage |
+|-----|-----------------|---------|
+| 1   | 3V3_FMU        | 3.3V    |
+| 2   | USART3_TX_DEBUG | 3.3V    |
+| 3   | USART3_RX_DEBUG | 3.3V    |
+| 4   | FMU_SWDIO      | 3.3V    |
+| 5   | FMU_SWCLK      | 3.3V    |
+| 5   | SPI6_SCK_EXTERNAL1      | 3.3V    |
+| 5   | NFC_GPIO      | 3.3V    |
+| 5   | PD15      | 3.3V    |
+| 5   | FMU_NRST      | 3.3V    |
+| 6   | GND            | GND     |
+
+
+## UART Mapping
+
+|Name|Function|
+|:-|:-|
+|SERIAL0|USB (Internal)|
+|SERIAL1|UART7 (Telem1)|
+|SERIAL2|UART5 (Internal)|
+|SERIAL3|USART1 (GPS1)|
+|SERIAL4|UART4 (ESC Telem)|
+|SERIAL6|USART6 (RC Input)|
+
+All UARTS support DMA. Any UART may be re-tasked by changing its protocol parameter.
+
+Ports marked _Internal_ are connected directly to the Raspberry Pi.
+
+## RC Input
+
+RC input is configured on the RX6 (UART6_RX) pin. It supports all RC protocols except PPM. See :ref:`Radio Control Systems <common-rc-systems>` for details for a specific RC system. :ref:`SERIAL6_PROTOCOL<SERIAL6_PROTOCOL>` is set to “23”, by default, to enable this.
+
+- SBUS/DSM/SRXL connects to the RX6 pin.
+- FPort requires connection to TX6 and :ref:`SERIAL6_OPTIONS<SERIAL2_OPTIONS>` be set to “7”.
+- CRSF also requires a TX6 connection, in addition to RX6, and automatically provides telemetry. Set :ref:`SERIAL6_OPTIONS<SERIAL6_OPTIONS>`
+- SRXL2 requires a connection to TX6 and automatically provides telemetry. Set :ref:`SERIAL6_OPTIONS<SERIAL6_OPTIONS>` to “4”. =3.
+
+## Battery Monitoring
+The board has an integrated INA226 digital power monitor. The board supports up to 6S LiPo batteries.
+
+The default battery parameters are:
+
+- BATT_MONITOR 21
+
+## Compass
+This autopilot has a built-in compass. The compass is the IIS2MDC
+
+## Motor Output
+
+All outputs are capable of PWM and DShot. Motors 1-4 are capable of Bidirectional-DSHOT. All outputs in the motor groups below must be either PWM or DShot:
+- Motors 1-4  Group1 (TIM5)
+- Motors 5-6  Group2 (TIM4)
+- Motors 7-8  Group3 (TIM12)
+
+## Loading Firmware
+
+Firmware can be loaded directly with SWD using the 10pin DEBUG port. Alternatively you can update the firmware over the USB interface which is connected directly to the Raspberry Pi. You can use the [uploader.py](https://github.com/ArduPilot/ardupilot/blob/master/Tools/scripts/uploader.py) flashing script from the Pi, or use the [software tools providied by ARK](https://github.com/ARK-Electronics/ARK-OS).
```

Comment:
ArduPilot can change behaviour if you have USB connected.  I don't think we enumerate those ways, but at least these:
 - we dim the notify LEDs
 - we can change the way we do logging (log-disarmed-but-not-if-usb-attached)

tridge has also described situations where the stm32 might have very high pauses in the processing ArduPilot sees based on USB traffic IIUC.  Probably want to understand that problem before using this design for flying vehicles.

Reply:
Thanks for the heads up. As always I appreciate the speedy reviews! Are you suggesting the USB activity can interfere with the control pipeline scheduling? Should we ask the man himself @tridge? We have the same direct USB connection on our Jetson PAB carrier with people currently flying Ardupilot

Reply:
No you can't connect the board to the GCS directly with USB. Instead the user should run mavproxy or mavlink-router on the Pi and connect to the GCS via Ethernet or WiFi. 

The FC USB is not exposed on a connector and is instead routed on the PCB to connect directly to the Pi via USB switch.